### DISCUSSION TITLE
[Bug] When querying topn, there seems to be a problem with the indicator aggregation calculation method.

### DISCUSSION THREAD

**Discussion by bbcdabao** [apache/skywalking#11956] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

When using related interfaces such as getEndpointTopN or getServiceTopN
it will not select the correct calculation method based on the indicator statistics method, but a hard-coded avg
The effect is as follows

I added an indicator in core.oal like ...
endpoint_count = from(Endpoint.*).count();   Note that it is count  function.
and when you call getEndpointTopN used endpoint_count as param , it used avg aggregation
![image](https://github.com/apache/skywalking/assets/45411947/ee7b7c3c-7cae-40f0-a23f-d92fcc239cba)
As shown above
It was called 248 times during this time, but getEendpointTopN got an average of 16 times, which I think should be the total calculated based on the metric rather than the tie value

### What you expected to happen

It was called 248 times during this time, but getEendpointTopN got an average of 16 times, which I think should be the total calculated based on the metric rather than the tie value

### How to reproduce

(1) Define a count aggregated indicator in core.oal, for example:endpoint_count = from(Endpoint.*).count(); note is count not avg
(2) Used  getXXXTopN interface ,and the return value is average 

### Anything else

There seems to be a problem with the storage as plug-in implementation. For example
![image](https://github.com/apache/skywalking/assets/45411947/667a7362-3c39-4b01-a1f0-1215a451631b)
The implementation of an interface as shown above
AggregationQueryEsDAO implementation
![image](https://github.com/apache/skywalking/assets/45411947/c17c7567-0514-4cdc-afcf-265d9b6217df)
It seems that the average value is used for aggregation.

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
I am following your examples, OAL is for calculation, not relative to query actually.
You are mixing two things together.

TopN query is applying for data after OAL process.

**Reply by bbcdabao** [reply]
Yes , TopN is query after OAL process ,They are also two independent processes , but my means TopN query is also consider what kind of aggregation to use。  I am modify like this 
![image](https://github.com/apache/skywalking/assets/45411947/758389a0-36ac-4451-90f9-d626d4fd8ee9)
I modified it as shown above
AggregationQueryEsDAO.sortMetrics  modify this function  and then TopN query  returned result is correct.
Thank you for your patient reply
Maybe there is something wrong with my understanding. Maybe everything is sorted according to the average value.

**Reply by wu-sheng** [reply]
Do you mean Elasticsearch aggregate implementation should follow declared function?
If so, yes. You could submit a pull request to fix.

**Reply by bbcdabao** [reply]
Yes, at least both cases for sum and avg supporters ，my good frist issue,  thank you

**Reply by wu-sheng** [reply]
Go ahead. Single value query is replaced by mqe for most cases.
But, agree, keeping aligned is good.

**Reply by wu-sheng** [reply]
You could change this locally, but there could be an argument that avg is a duration based top N list.

**Reply by bbcdabao** [reply]
ok thank you

**Reply by wu-sheng** [reply]
So, the conclusion is, as single value query protocol is deprecated, I am going to remove the value column function, and make them all use `avg` without configurable. 
For your scenarios, you could do a MQE query, and use `sum` functions in the query stage.

**Reply by bbcdabao** [reply]
Ok  thank you ,  I  learned

**Reply by bbcdabao** [reply]
I learned MQE syntax and it's very powerful and flexible .    some times like  sum(endpoint_count)

**Reply by wu-sheng** [reply]
Yes, That it is. If you want to sum or anything, it is there for you.

**Comment by wu-sheng** [comment]
We will follow this on https://github.com/apache/skywalking/issues/11960
